### PR TITLE
[SPARK-32476][CORE] ResourceAllocator.availableAddrs should be deterministic

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to make `ResourceAllocator.availableAddrs` deterministic.

### Why are the changes needed?

Currently, this function returns indeterministically due to the underlying `HashMap`. So, the test case itself is creating a list `[0, 1, 2]` initially, but ends up with comparing `[2, 1, 0]`.

Not only this happens in the 3.0.0, but also this causes UT failures on Scala 2.13 environment.

### Does this PR introduce _any_ user-facing change?

Yes, but this fixes the in-deterministic behavior.

### How was this patch tested?

- Scala 2.12: This should pass the UT with the modified test case.
- Scala 2.13: This can be tested like the following (at least `JsonProtocolSuite`)

```
$ dev/change-scala-version.sh 2.13
$ build/mvn test -pl core --am -Pscala-2.13 -Dtest=none -DwildcardSuites=org.apache.spark.deploy.JsonProtocolSuite
```

**BEFORE**
```
*** 2 TESTS FAILED ***
```

**AFTER**
```
All tests passed.
```

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @tgravescs and @Ngone51 FYI

**Comment by dongjoon-hyun** [comment]
Thanks, @HyukjinKwon .
Also, cc @srowen

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by Ngone51** [comment]
The test only fail under Scala 2.13? Does it mean there are changes in the underlying `HashMap` that affect the order of key?

The fix itself looks good to me.

File: core/src/main/scala/org/apache/spark/resource/ResourceAllocator.scala [hunk 1]
```diff
@@ -56,7 +56,7 @@ trait ResourceAllocator {
   def availableAddrs: Seq[String] = addressAvailabilityMap
     .flatMap { case (addr, available) =>
       (0 until available).map(_ => addr)
-    }.toSeq
+    }.toSeq.sorted
```
Comment (tgravescs) [inline core/src/main/scala/org/apache/spark/resource/ResourceAllocator.scala:59]: we should do assignedAddrs as well to keep them consistent. There are tests using it but I didn't see

**Comment by dongjoon-hyun** [comment]
Thanks, @Ngone51 and @tgravescs . Sure. I'll update it too.

**Comment by dongjoon-hyun** [comment]
@tgravescs , `assignedAddrs` is updated, too.

**Comment by dongjoon-hyun** [comment]
cc @srowen

**Review by tgravescs — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Thanks, @tgravescs !

**Review by srowen — COMMENTED** [review]
Seems OK pending tests

**Comment by dongjoon-hyun** [comment]
Thanks, @srowen . Yep. Pending tests.

**Comment by dongjoon-hyun** [comment]
Jenkins passed. Merged to master.

**Pull request closed without merging**